### PR TITLE
Add functionality for next and prev buttons.

### PR DESCRIPTION
**Pull request by wudi96**
Just added some functionality for next and prev buttons. Includes event-handler, options and enabling-disabling of the button in activatePanel.

### CODE REVIEW

**Comment by guidobouman** [comment]
Hey,

First of all: good PR!

I do think we need a couple of improvements:
1. Use spaces, not tabs.
2. I think we should merge the keyboard & button navigations.
3. While we're merging the navigations, I vote for removing the `enabled` property on the navigation block. Just define the keys and/or buttons, and navigation will work. I would simplify the implementation.

I think something like the following would do:
```
{
  navigation: {
    keys: {
      nextKey: 40,
      previousKey: 38
    },
    buttons: {
      $nextButton: false,
      $prevButton: false
    },
    wrapAround: true
}
```

The `wrapAround` option should work the same for both keyboard and button navigation.

What's your opinion on this?

**Comment by wudi96** [comment]
Hey,

yeah looks great. As I'm just a pupil with experiences -> 0 it's up to you.
But I think the enabling and disabling functionality is quit important. Not sure but I think it's quite good for accessibility to disable buttons which doesn't have a function currently.

Cheers,
Martin

**Comment by guidobouman** [comment]
Oh, yeah, disabling the buttons when at an end is a must. I think I'll do it through at least a class (and maybe an aria attribute as well).

But the plugin can figure out when you provide a button of key, and that should be enough for the plugin to enable or disable keyboard and or button navigation in the code. That's not something that you should have to think about in the implementation.

**Comment by guidobouman** [comment]
By the way, do you want to refactor your PR to cohere to my suggestions, or want me to pick this up later?

**Comment by wudi96** [comment]
Hi,
ah yeah sry, I missunderstood the thing with the enabled property ( it's 17 days ago ;) ).

I can try to refactor it, but probably not today. Not sure if I can meet all your quality goals, but we'll see.

Cheers,
Martin

**Comment by wudi96** [comment]
Have a look.

Bad thing: we're loosing compatibility.

File: jquery.panelSnap.js [hunk 1]
```diff
@@ -377,6 +413,35 @@ if ( typeof Object.create !== 'function' ) {
         var $itemToActivate = $(itemSelector, self.options.$menu);
         $itemToActivate.addClass('active');
       }
+      
+      var nav = self.options.navigation;
+      
+      if(!nav.wrapAround) {
+      	
+      	var $panels = self.getPanel();
+        var index = $panels.index(self.getPanel('.active'));
```
Comment (guidobouman) [inline jquery.panelSnap.js]: :rotating_light:  Leading space. :rotating_light:

File: jquery.panelSnap.js [hunk 2]
```diff
@@ -282,17 +290,45 @@ if ( typeof Object.create !== 'function' ) {
...
+        return;
+      }
+      
+      self.snapTo('next', self.options.navigation.wrapAround);
+    	
+    },
+    
+    capturePrevClick: function(e) {
+    	
+    	var self = this;
```
Comment (guidobouman) [inline jquery.panelSnap.js]: :rotating_light: Tabs. :rotating_light:

File: jquery.panelSnap.js [hunk 3]
```diff
@@ -282,17 +290,45 @@ if ( typeof Object.create !== 'function' ) {
...
           e.preventDefault();
           self.snapTo('next', nav.wrapAround);
           break;
       }
 
     },
+    
+    captureNextClick: function(e) {
+    	
+    	var self = this;
```
Comment (guidobouman) [inline jquery.panelSnap.js]: :rotating_light: Tabs :rotating_light:

**Comment by guidobouman** [comment]
OK, apart from the whitespace errors this looks great. Fix that and I'll merge this in. Great work.

The big advantage of still being in `v0.x` is that I can still break the API whenever I please. Once I enter `v1.x` I'll take this into account. For now: this seems to be a better approach.

File: README.md [hunk 4]
```diff
@@ -51,10 +51,15 @@ var options = {
...
-    nextPanelKey: 40,
-    previousPanelKey: 38,
+  navigation: {
+    keys: {
+      nextKey: 40,
+       prevKey: 38,
+    },
+    buttons: {
+      $nextButton: $('#nextButton'),
+      $prevButton: $('#prevButton'),
```
Comment (guidobouman) [inline README.md]: This should represent the default value. By default the navigation is disabled which means they should be false.

File: jquery.panelSnap.js [hunk 5]
```diff
@@ -535,10 +600,15 @@ if ( typeof Object.create !== 'function' ) {
...
     easing: 'linear',
     offset: 0,
-    keyboardNavigation: {
-      enabled: false,
-      nextPanelKey: 40,
-      previousPanelKey: 38,
+    navigation: {
+      keys: {
+        nextKey: 40,
+        prevKey: 38
```
Comment (guidobouman) [inline jquery.panelSnap.js]: These should be false or null by default. Otherwise keyboard navigation will be enabled by default.

**Comment by guidobouman** [comment]
OK, found a small issue on the defaults & their documentation. Once that's tackled I can merge this is.

**Comment by guidobouman** [comment]
Actually, I have some time on my hands, I'll pick up the nitty gritty details. Make sure you sweat the details next time. ;)

Thanks for your great contribution though!

**Comment by wudi96** [comment]
Oh great. Thanks a lot for merging it :)

Was my first time with github - didn't know that people don't like tabs - but learned my lesson (changed the setting in my editor).
You're welcome.
Cheers

**Comment by guidobouman** [comment]
Just added an .editorconfig file to prevent this from happening again. ;)

**Pull request merged**